test/irmin-bench/replay.ml

```diff
@@ -0,0 +1,92 @@
+open! Import
+
+let test_dir = Filename.concat "_build" "test-pack-trace-replay"
+
+module Conf = struct
+  let entries = 32
+  let stable_hash = 256
+end
+
+module Store = struct
+  type store_config = string
+
+  open Tezos_context_hash_irmin.Encoding
+
+  module V1 = struct
+    let version = `V1
+  end
+
+  module Maker = Irmin_pack.Maker_ext (V1) (Conf) (Node) (Commit)
+  module Store = Maker.Make (Metadata) (Contents) (Path) (Branch) (Hash)
+
+  let create_repo store_dir =
+    let conf = Irmin_pack.config ~readonly:false ~fresh:true store_dir in
+    let* repo = Store.Repo.v conf in
+    let on_commit _ _ = Lwt.return_unit in
+    let on_end () = Lwt.return_unit in
+    let pp _ = () in
+    Lwt.return (repo, on_commit, on_end, pp)
+
+  include Store
+end
+
+module Replay = Irmin_traces.Trace_replay.Make (Store)
+
+let rec repeat = function
+  | 0 -> fun _f x -> x
+  | n -> fun f x -> f (repeat (n - 1) f x)
+
+(** The current working directory depends on whether the test binary is directly
+    run or is triggered with [dune exec], [dune runtest]. We normalise by
+    switching to the project root first. *)
+let goto_project_root () =
+  let cwd = Fpath.v (Sys.getcwd ()) in
+  match cwd |> Fpath.segs |> List.rev with
+  | "irmin-bench" :: "test" :: "default" :: "_build" :: _ ->
+      let root = cwd |> repeat 4 Fpath.parent in
+      Unix.chdir (Fpath.to_string root)
+  | _ ->
+      ()
```

Comment:
Somewhat unrelated, it's possible to generalise this w/ the `__FILE__` macro, so we should maybe have a test-utils private lib some day.